packages/@aws-cdk/aws-stepfunctions-tasks/lib/lambda/invoke.ts

```diff
@@ -0,0 +1,149 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { getResourceArn, TaskStateConfig, taskStateJson, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for invoking a Lambda function with LambdaInvoke
+ */
+export interface LambdaInvokeProps extends sfn.TaskStateBaseProps {
+
+  /**
+   * Lambda function to invoke
+   */
+  readonly lambdaFunction: lambda.IFunction;
+
+  /**
+   * The JSON that will be supplied as input to the Lambda function
+   *
+   * @default - The state input (JSON path '$')
+   */
+  readonly payload?: sfn.TaskInput;
+
+  /**
+   * Invocation type of the Lambda function
+   *
+   * @default InvocationType.REQUEST_RESPONSE
+   */
+  readonly invocationType?: LambdaInvocationType;
+
+  /**
+   * Client context to pass to the function
+   *
+   * @default - No context
+   */
+  readonly clientContext?: string;
+
+  /**
+   * Version or alias of the function to be invoked
+   *
+   * @default - No qualifier
+   */
+  readonly qualifier?: string;
+}
+
+/**
+ * Invoke a Lambda function as a Task
+ *
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/connect-lambda.html
+ */
+export class LambdaInvoke extends sfn.TaskStateBase {
+
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS: sfn.IntegrationPattern[] = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+  private readonly metricsConfig: sfn.TaskMetricsConfig;
+  private readonly policies: iam.PolicyStatement[];
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: LambdaInvokeProps) {
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, LambdaInvoke.SUPPORTED_INTEGRATION_PATTERNS);
+
+    // TODO - why not just make the task token an input to the task?
```

Comment:
User gets to pick where in the Payload it is passed. Might look like this:

```
payload: TaskInput.fromObject({
  Foo: "Do a foo",
  Bar: Context.taskToken(),
})
```

You can't really generate that for them.

Reply:
fair point, getting rid of the TODO